x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -117,6 +120,290 @@ export const installTransform = async (
+  installNameSuffix: string,
+  transformPaths: string[],
+  esClient: ElasticsearchClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  logger: Logger,
+  esReferences: EsAssetReference[] = [],
+  previousInstalledTransformEsAssets: EsAssetReference[] = []
+) => {
+  let installedTransforms: EsAssetReference[] = [];
+  if (transformPaths.length > 0) {
+    const {
+      indexTemplatesRefs,
+      componentTemplatesRefs,
+      transformRefs,
+      transforms,
+      destinationIndexTemplates,
+      transformsSpecifications,
+    } = processTransformAssetsPerModule(installablePackage, installNameSuffix, transformPaths);
+    // get and save refs associated with the transforms before installing
+    esReferences = await updateEsAssetReferences(
+      savedObjectsClient,
+      installablePackage.name,
+      esReferences,
+      {
+        assetsToAdd: [...indexTemplatesRefs, ...componentTemplatesRefs, ...transformRefs],
+        assetsToRemove: previousInstalledTransformEsAssets,
+      }
+    );
+
+    // create index templates and component templates
+    await Promise.all(
+      destinationIndexTemplates
+        .map((destinationIndexTemplate) => {
+          const customMappings = transformsSpecifications
+            .get(destinationIndexTemplate.transformModuleId)
+            ?.get('mappings');
+          const customTemplateName = `${destinationIndexTemplate.installationName}${USER_SETTINGS_TEMPLATE_SUFFIX}`;
+          const packageTemplateName = `${destinationIndexTemplate.installationName}${PACKAGE_TEMPLATE_SUFFIX}`;
+          if (destinationIndexTemplate !== undefined || customMappings !== undefined) {
+            return installComponentAndIndexTemplateForDataStream({
+              esClient,
+              logger,
+              componentTemplates: {
+                ...(customMappings
+                  ? {
+                      [customTemplateName]: {
+                        template: {
+                          mappings: customMappings,
+                        },
+                        _meta: destinationIndexTemplate._meta,
+                      },
+                    }
+                  : {}),
+                ...(destinationIndexTemplate.template
+                  ? {
+                      [packageTemplateName]: {
+                        template: {
+                          settings: destinationIndexTemplate.template.settings,
+                          mappings: destinationIndexTemplate.template.mappings,
+                        },
+                        _meta: destinationIndexTemplate._meta,
+                      },
+                    }
+                  : {}),
+              },
+              indexTemplate: {
+                templateName: destinationIndexTemplate.installationName,
+                // @ts-expect-error Index template here should not contain data_stream property
+                // as this template is applied to only an index and not a data stream
```

Comment:
To be clear, this is shortcoming in our typings for this function, right? 

https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/types/models/epm.ts#L505-L515

`data_stream` is required here, but making it nullable would probably cause a bunch of other type errors elsewhere.

Reply:
That's correct! I don't see the need to update the type for [IndexTemplate](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/types/models/epm.ts#L528) interface sice it's a one-off but I have updated the expect-error comment [here](https://github.com/elastic/kibana/pull/140046/commits/5a98445cfc4c630186c503ad2cfca7893628b6a8) for better clarity.